lumigator/backend/backend/services/jobs.py

```diff
@@ -176,15 +177,21 @@ def _add_dataset_to_db(self, job_id: UUID, request: JobInferenceCreate, s3: S3Fi
         loguru.logger.info("Adding a new dataset entry to the database...")
 
         # Get the dataset from the S3 bucket
-        result_key = self._get_results_s3_key(job_id)
-        with s3.open(f"{settings.S3_BUCKET}/{result_key}", "r") as f:
-            # Validate that the output file adheres to the expected inference output schema
-            results = InferenceJobOutput.model_validate(json.loads(f.read()))
+        results = self._validate_results(job_id, s3)
+
+        # make sure the artifacts are present in the results
+        if not all(
+            key in results.artifacts for key in ["examples", "ground_truth", request.output_field]
+        ):
+            raise ValueError("Missing required fields in the evaluation results.")
 
-        # define the fields we want to keep from the results JSON
-        # and build a CSV file from it as a BytesIO object
-        fields = ["examples", "ground_truth", request.output_field]
-        bin_data = self._results_to_binary_file(results.model_dump(), fields)
+        dataset_to_save = {
+            "examples": results.artifacts["examples"],
+            "ground_truth": results.artifacts["ground_truth"],
+            request.output_field: results.artifacts[request.output_field],
+        }
```

Comment:
Hum. I'd add an additional layer under `artifacts`. What if a job produces more than one dataset?
Something like:
```suggestion
        dataset_to_save = {
            "examples": results.artifacts[output_artifact]["examples"],
            "ground_truth": results.artifacts[output_artifact]["ground_truth"],
            request.output_field: results.artifacts[output_artifact][request.output_field],
        }
```
Maybe this `output_artifact` is a constant at this stage, or a well known value, so it doesn't need to be specified in the job itself.

Reply:
`What if a job produces more than one dataset?` I agree that we should cover this corner case. However, at the moment we do assume that a job only produces a single dataset. Whenever we want to support the use case where a job is saving multiple datasets, we'll need to refactor a bunch of code. I'd rather not try to support a new feature in this PR, but I think it's a good idea to track that we should do this. Created https://github.com/mozilla-ai/lumigator/issues/835 to keep track of this.